tests/rootfs.sh

```diff
@@ -0,0 +1,149 @@
+TEST_IMAGE_FILE_PATH="${ROOTFS_DIRECTORY}/tmp/test.img"
+TEST_EMPTY_MTAB_FILE="${ROOTFS_DIRECTORY}/etc/mtab"
+
+test_setup()
+{
+    if [ ! -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        cp "$(command -v qemu-arm-static)" "${ROOTFS_DIRECTORY}/usr/bin/"
+    fi
+
+    dd if=/dev/zero of="${TEST_IMAGE_FILE_PATH}" bs=2M count=64 >/dev/null 2>&1 || return ${ERROR}
+    touch "${TEST_EMPTY_MTAB_FILE}" >/dev/null 2>&1 || return ${ERROR}
+    mkfs.ext4 "${TEST_IMAGE_FILE_PATH}" >/dev/null 2>&1 || return ${ERROR}
+    mount -o bind /dev/ "${ROOTFS_DIRECTORY}/dev/" >/dev/null 2>&1 || return ${ERROR}
+}
+
+test_destroy()
+{
+    umount -l "${ROOTFS_DIRECTORY}/dev"
+
+    if [ -f "${TEST_IMAGE_FILE_PATH}" ]
+    then
+        rm -f ${TEST_IMAGE_FILE_PATH}
+    fi
+
+    if [ -f "${TEST_EMPTY_MTAB_FILE}" ]
+    then
+        rm -f "${TEST_EMPTY_MTAB_FILE}"
+    fi
+    if [ -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        rm -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static"
+    fi
+}
+
+run_test()
+{
+    if ! test_setup
+    then
+        printf "Cannot run tests, unable to complete test setup\\n"
+        test_destroy
+        exit ${ERROR}
+    fi
+
+    if "$1"
+    then
+        printf "Run: %s - OK\\n" "${1}"
+    else
+        printf "Run: %s - ERROR\\n" "${1}"
+    fi
+    test_destroy
+}
+
+test_execute_resize2fs()
+{
+    ( chroot "${ROOTFS_DIRECTORY}" /sbin/resize2fs /tmp/test.img >/dev/null 2>&1 && return "${OK}" ) || return "${ERROR}"
+}
+
+test_execute_fdisk()
+{
+    ( chroot "${ROOTFS_DIRECTORY}" /sbin/fdisk -v >/dev/null 2>&1 && return "${OK}" ) || return "${ERROR}"
+}
+
+test_execute_mount()
+{
+   ( chroot "${ROOTFS_DIRECTORY}" /bin/mount >/dev/null 2>&1 && return "${OK}" ) || return "${ERROR}"
+}
+
+test_execute_rsync()
+{
+    ( chroot "${ROOTFS_DIRECTORY}" /usr/bin/rsync -h >/dev/null 2>&1 && return "${OK}" ) || return "${ERROR}"
+}
+
+test_execute_busybox()
+{
+    ( chroot "${ROOTFS_DIRECTORY}" /bin/busybox --help >/dev/null 2>&1 && return "${OK}" ) || return "${ERROR}"
+}
+
+test_execute_mkfs_ext4()
+{
+    ( chroot "${ROOTFS_DIRECTORY}" /sbin/mkfs.ext4 /tmp/test.img >/dev/null 2>&1 && return "${OK}" ) || return "${ERROR}"
+}
+
+test_execute_mkfs_f2fs()
+{
+    ( chroot "${ROOTFS_DIRECTORY}" /sbin/mkfs.f2fs /tmp/test.img >/dev/null 2>&1 && return "${OK}" ) || return "${ERROR}"
+}
+
+usage()
+{
+cat <<-EOT
+Usage:   ./${MY_NAME} -r path to bootstrapped rootfs [ | -h ]
+-h   Print usage
+-r   Path to the bootstrapped rootfs that needs to be tested.
+NOTE: This script should be run as root
+EOT
+}
+
+if [ "$(id -u)" != "0" ]
+then
+    printf "Make sure this script is run as root\\n"
+    usage
+    exit
+fi
+
+while getopts ":hr:" options
+do
+  case "${options}" in
+    h )
+      usage
+      exit ${OK}
+      ;;
+    r )
+      ROOTFS_DIRECTORY="${OPTARG}"
+      TEST_IMAGE_FILE_PATH="${ROOTFS_DIRECTORY}/tmp/test.img"
+      TEST_EMPTY_MTAB_FILE="${ROOTFS_DIRECTORY}/etc/mtab"
```

Comment:
this is the final notion of the mtab file, but i don't see it being _used_ anywhere. I suppose some intermediate test fails with this missing? As it can also be empty, can we not bind mount our current system one?

Reply:
Yess  both mkfs.f2fs and resize2fs require the existence of this file. 
Bind mounting is a good idea so that the tools can check if the file is mounted, but then it is not required for the test because they succeed anyway and it requires more loc. 
But then again, maybe this is an error of the tool and therefore in a future version this will be fixed, so gonno bind mount it anyway.